### ISSUE TITLE
[BUG] Graph(style='height=100%') does not work

### ISSUE THREAD

**Issue by orenbenkiki** [plotly/dash#857]
**PIP**

```
dash (1.0.2)
dash-bootstrap-components (0.7.0)
dash-core-components (1.0.0)
dash-daq (0.1.7)
dash-html-components (1.0.0)
dash-renderer (1.0.0)
dash-table (4.0.2)
```

**Describe the bug**

Writing: `Graph(..., style=dict(height='100%'))`

**Expected behavior**

The graph should fill the full height of the `div`, which will obey normal CSS rules, that is, span all the available space in the layout.

**Actual behavior**

The graph has a height of 100 pixels.

**Discussion**

The default graph height is 450 pixels.
I can create a graph with a fixed height in pixels using `height='800px'`.
I can create a graph that spans _all_ the visible display, and reacts to window size changes, using `height='100vh'`.
But I *can't* create a graph that spans the _remaining_ visible display (given some other HTML elements above and/or below the graph), and reacts to window size changes, using `height='100%'`.

Obviously dash *is* capable of responding to window size changes. Otherwise, `height='100vh`' would not work. It therefore seems to be a bug that `height='100%'` does not work.

**Comment by orenbenkiki** [comment]
Here is an example showing the problem:

```python
import dash
import dash_core_components as dcc
import dash_html_components as html

app = dash.Dash()
app.layout = html.Div(
    id='body',
    style={
        'display': 'flex',
        'flex-flow': 'column',
        'background-color': 'green',
        'height': '100vh',
        'width': '100%'
    },
    children=[
        html.Div(
            id='header',
            children=[
                html.H1('MWE'),
            ]
        ),
        html.Div(
            id='expand',
            style={
                'flex-grow': 1,
                'height': '100%',
                'width': '100%',
                'background-color': 'blue'
            },
            children=[
                dcc.Graph(figure=dict(layout=dict(autosize=True))),
            ]
        ),
    ]
)
app.run_server(debug=True)
```

**Comment by orenbenkiki** [comment]
Although the sample example above sort-of-works, in my real-life application it doesn't :-( I have a complex layout using nested flex boxes, specifically a top bar, a side bar, and a main display region which tries to span the remaining space.

Things I have seen:

- The _initial_ size of the graph is set to 100px * 100px. Any resize event (e.g. if I resize the window) causes the graph to resize to the full size.

- Trying to show a heatmap of 100x * 100px causes "Something went wrong with axis scaling" because the computation of the colorbar scale becomes confused due to the cramped space.

Bottom line: the automatic resize of the graph when the container is at width and height of 100% is too fragile to work for me. I am working around the issue by giving some elements fixed sizes and setting the graph size to something horrible like `calc(100vw - 400px)`. This does work at the cost of using absolute sizes for elements I'd much rather just have their "natural" size :-(
